posts/2018-11-02-semirings.md

```diff
@@ -0,0 +1,435 @@
+---
+layout: post
+title: A tale on Semirings
+
+meta:
+  nav: blog
+  author: lukajcb
+  pygments: true
+
+tut:
+  scala: 2.12.7
+  binaryScala: "2.12"
+  scalacOptions:
+    - "-Ypartial-unification"
+    - "-language:higherKinds"
+    - "-language:implicitConversions"
+    - "-language:experimental.macros"
+  dependencies:
+    - com.github.mpilquist::simulacrum:0.14.0
+  plugins:
+    - org.scalamacros:paradise_2.12.7:2.1.0
+---
+
+
+
+Ever wondered why sum types are called sum types?
+Or maybe you've always wondered why the `<*>` operator uses exactly these symbols?
+And what do these things have to do with Semirings?
+Read this article and find out!
+
+
+----
```

Comment:
Please don't use these horizontal rulers. Instead, emphasize the introductory text (italics, for example).